Model/BalancepayMethod.php

```diff
@@ -229,13 +238,19 @@ public function __construct(
     /**
      * Check whether payment method can be used
      *
-     * @param \Magento\Quote\Api\Data\CartInterface|null $quote
+     * @param CartInterface|null $quote
      * @return bool
      * @deprecated 100.2.0
      */
-    public function isAvailable(\Magento\Quote\Api\Data\CartInterface $quote = null)
+    public function isAvailable(CartInterface $quote = null)
     {
-        if ($quote && $quote->isMultipleShippingAddresses()) {
+        $isCustomerGroupAllowed = false;
+        $currentCustomerGroup = $this->customerSession->getCustomer()->getGroupId();
+        $allowedCustomerGroups = $this->balancepayConfig->getAllowedCustomerGroups();
+        if (!empty($allowedCustomerGroups)) {
+            $isCustomerGroupAllowed = in_array($currentCustomerGroup, explode(',', $allowedCustomerGroups));
+        }
```

Comment:
```suggestion
        $currentCustomerGroup = $this->customerSession->getCustomer()->getGroupId();
        $allowedCustomerGroups = explode(',',$this->balancepayConfig->getAllowedCustomerGroups());
        $isCustomerGroupAllowed = in_array($currentCustomerGroup, $allowedCustomerGroups);
```

Reply:
I would consider returning an array from `getAllowedCustomerGroups()` instead of string